### PR TITLE
Backport docs migration

### PR DESCRIPTION
**Pull request by willhickey**
#### Problem
Docs are deployed from the beta branch. The docs migration is incomplete until Nick's changes are applied to beta.

#### Summary of Changes
Backport:
#34096
#34403
#34565

Fixed merge conflicts. 

Most of merge conflicts were just whitespace so I kept the v1.17 version. But I applied the master link changes from #34380. I think this is the correct behavior but since the channels will be advancing when mainnet-beta adopts v1.17 (hopefully within 2 weeks) any mistakes will be irrelevant soon.

### CODE REVIEW

File: docs/src/developing/on-chain-programs/developing-rust.md [hunk 1]
```diff
@@ -52,14 +52,14 @@ For example:
   on Rand](#depending-on-rand).
 - Crates may overflow the stack even if the stack overflowing code isn't
   included in the program itself. For more information refer to
-  [Stack](./faq.md#stack).
+  [Stack](https://docs.solana.com/developing/on-chain-programs/faq#stack).
```
Comment (nickfrosty) [inline docs/src/developing/on-chain-programs/developing-rust.md:55]: This should be:
```
[Stack](https://solana.com/docs/programs/faq#stack).
```

File: docs/src/developing/on-chain-programs/developing-rust.md
Comment (nickfrosty) [inline docs/src/developing/on-chain-programs/developing-rust.md:1]: This file should be deleted per https://github.com/solana-labs/solana/commit/676e80c80a914c6f41a4f5d12f419d35a8047dd8

File: docs/src/developing/versioned-transactions.md
Comment (nickfrosty) [inline docs/src/developing/versioned-transactions.md:1]: This file should be deleted per https://github.com/solana-labs/solana/commit/676e80c80a914c6f41a4f5d12f419d35a8047dd8

**Review by nickfrosty — CHANGES_REQUESTED** [review]
I noticed 2 files did not get removed with this backport

**Comment by willhickey** [comment]
Since the release channels have advanced we no longer need this PR. The changes in this PR were merged to master before v1.18 was branched. Now that v1.18 is the beta branch it will feed https://docs.solanalabs.com/. The docs in v1.17 don't get published anywhere.

**Pull request closed without merging**